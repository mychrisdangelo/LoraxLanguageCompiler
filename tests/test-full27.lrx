int main()
{
	tree <int> t(2);
	tree <int> s(2);
	tree <int> u(2);
	string v;
	string w;
	int i;

	v = "abcdefg";
	w = "hijklmn";

	v%0%0@ = 'Z';
	v%0%0%0 = w;

	print(v, '\n');

	t = 1[2[-101, 102], 5];
	s = 6[7[8,9], 10[]];
	u = 1001[1002[1003, 1004], 1005[]];


	// print("s@ = ", s@, "\n,s@ = ", u%0%0@, "\n, s@ + t%0%1@ = ", s@ + t%0%1@, "\n");

	t%0@ = 201; //	t = 1[201[-101, 102], 5];
	s = t%1 = u%0; //	t = 1[2[-101, 102], 1002[1003, 1004]];
	print("s = ", s, "\nt = ", t, "\nu = ", u, '\n');

	t = 1[2[-101, 102], 5];
	s = 6[7[8,9], 10[]];


	for (i = 0; i < 2; i = i+1) {
		t%0%i@ = i;
	}
	print(t, '\n', s);
}