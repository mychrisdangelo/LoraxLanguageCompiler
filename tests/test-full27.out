abZhijklmn
s = 1002[1003[null,null],1004[null,null]]
t = 1[201[-101[null,null],102[null,null]],1002[1003[null,null],1004[null,null]]]
u = 1001[1002[1003[null,null],1004[null,null]],1005[null,null]]
1[2[0[null,null],1[null,null]],5[null,null]]
6[7[8[null,null],9[null,null]],10[null,null]]